src/ChainSafe.Gaming/MultiCall/MultiCall.cs

```diff
@@ -0,0 +1,114 @@
+
+        private const string DefaultDeploymentAddress = "0xcA11bde05977b3631167028862bE2a173976CA11";
+        private readonly Contract multiCallContract;
+
+        public MultiCall(IContractBuilder builder, IChainConfig chainConfig, MultiCallConfig config)
+        {
+            if (chainConfig.ChainId != null && MultiCallDefaults.DeployedNetworks.Contains(chainConfig.ChainId))
+            {
+                multiCallContract = builder.Build(ContractAbi, DefaultDeploymentAddress);
+            }
+            else
+            {
+                if (chainConfig.ChainId != null && config.CustomNetworks.TryGetValue(chainConfig.ChainId, out var address))
+                {
+                    multiCallContract = builder.Build(ContractAbi, address);
+                }
```

Comment:
should we maybe throw an exception for an `else` here?

Reply:
Roger that

Reply:
Has this been updated? I still see it in changes